From https://github.com/joshmoore/zeroc-ice.git:

$ git log 42e01a5..d9463cb5
commit d9463cb539f494d88c724dfb8e6fa7e9ecb3a68d
Author: jmoore <josh@glencoesoftware.com>
Date:   Wed Aug 24 21:01:58 2011 +0200

    Fix 3.4.0 Patch #2
    
    Since 3.4.0 Patch #1 did not apply cleanly, patch #2
    was incorrectly applied.

commit 6519b0d5643f3a6738b1076d9f735b4f4aa1fa69
Author: jmoore <josh@glencoesoftware.com>
Date:   Wed Aug 24 20:51:35 2011 +0200

    Patch #2 for Ice 3.4.0: slice2cpp stream issue #2
    
    http://www.zeroc.com/forums/patches/4864-patch-2-ice-3-4-0-slice2cpp-stream-issue-2-a.html

diff --git a/cpp/src/slice2cpp/Gen.cpp b/cpp/src/slice2cpp/Gen.cpp
index b2a5b53..1beb938 100644
--- a/cpp/src/slice2cpp/Gen.cpp
+++ b/cpp/src/slice2cpp/Gen.cpp
@@ -321,6 +321,10 @@ Slice::Gen::generate(const UnitPtr& p)
             {
                 H << nl << "#include <" << s.substr(includePrefix.size()) << ">";
             }
+            else
+            {
+                H << "\n#include <Ice/StreamF.h>";
+            }
         }
     }
 
